src/App.tsx

```diff
@@ -15,48 +16,133 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+  let resultList;
+
+  switch (sortType) {
+    case SortType.ALPABET:
+      resultList = visibleGoods.sort((goodA, goodB) => (
+        goodA.localeCompare(goodB)
+      ));
+      break;
+
+    case SortType.LENGTH:
+      resultList = visibleGoods.sort((goodA, goodB) => (
+        goodA.length - goodB.length
+      ));
+      break;
+    default:
+      resultList = visibleGoods;
+      break;
+  }
+
+  if (isReversed) {
+    return resultList.reverse();
+  }
+
+  return resultList;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  // eslint-disable-next-line react/sort-comp
+  const sortByAlphabet = () => {
+    setSortType(SortType.ALPABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const makeReverse = () => {
+    setIsReversed(!isReversed);
```

Comment:
don't pass state value to setState function, you can pass function here (like functional setState, but in function returned from hook)